src/domains/issue/controller/issue.controller.ts

```diff
@@ -15,7 +17,9 @@ const issueRepo: IssueRepository = new IssueRepositoryImpl(db);
 const eventRepo: EventRepository = new EventRepositoryImpl(db);
 const userRepo: UserRepository = new UserRepositoryImpl(db);
 const projectRepo: ProjectRepository = new ProjectRepositoryImpl(db);
-const issueService: IssueService = new IssueServiceImpl(issueRepo, eventRepo, userRepo, projectRepo);
+const userProjectRoleRepo: UserProjectRoleRepository = new UserProjectRoleRepositoryImpl(db);
+const roleRepo: RoleRepository = new RoleRepositoryImpl(db);
+const issueService: IssueService = new IssueServiceImpl(issueRepo, eventRepo, userRepo, projectRepo, userProjectRoleRepo, roleRepo);
 
 issueRouter.post('/dev/:userId/project/:projectId', validateRequest(UserProjectParamsDTO, 'params'), validateRequest(DevOptionalIssueFiltersDTO, 'body'), async (req: Request<UserProjectParamsDTO, any, DevOptionalIssueFiltersDTO>, res: Response) => {
```

Comment:
As a suggestion, I think user roles (authorization) must be checked before the flow reaches the service. You can do it as a middleware and return a 403 code.

Reply:
Middleware created